tests/signature_tests/frequency_tests/test_traffic_freq_signatures.py

```diff
@@ -0,0 +1,140 @@
+import unittest
+import unittest.mock as um
+from scapy.layers.l2 import Ether
+from scapy.layers.inet import IP, UDP, TCP
+from src.anamoly_detection.frequency_signatures.traffic.traffic_layer_frequency_signature \
+    import TrafficLayerFrequencySignature
+from src.anamoly_detection.equation_parser import parse_equation
+
+root_path_test_data = "../../test_data/"
+
+
+class TestIPSignature(unittest.TestCase):
+    def setUp(self) -> None:
+        eq = parse_equation("1,2,3")    # f(x) = x + 2(x ^ 2) + 3
+        dev_eq = parse_equation("0")    # f(x) = 0
+        layer = "UDP"
+        self.signature = TrafficLayerFrequencySignature(eq, dev_eq, layer)
+        self.trigger_packet = Ether() / IP() / UDP(dport=80)
+        self.no_trigger_packet = Ether() / IP() / TCP(dport=80)
+
+    @um.patch("src.dashboard.alerts.alert.Alert.alert")
+    def test_acceptable_levels(self, mock_alert):
```

Comment:
This should be split up into multiple tests

Reply:
I mean it's definitely a long one, but why does it need to be split? I need to test that when I wrap around from 23 hours to 0 hours, the queue is shifted properly. I can rename it to `test_queue_wrap` to better represent the test, but all of the time shifts still need to all exist within the same test, otherwise the signature object will not be the same and the queue will not properly be tested.

Reply:
Ah gotcha, I misunderstood the need for the test. Maybe add a comment explaining?